auth_mellon_util.c

```diff
@@ -1835,7 +1957,130 @@ char *am_get_assertion_consumer_service_by_binding(LassoProvider *provider, cons
     return url;
 }
 
+/* Join array of strings with separator
+ *
+ * Parameters:
+ *  pool      memory allocation pool
+ *  strings   NULL terminated array of NULL terminated strings
+ *  separator string to insert between elements
+ *
+ * Example:
+ *  char *strings[] = {"one", "two", NULL};
+ *  am_str_join(pool, strings, ", ")
+ *
+ *  would return "one, two"
+ */
+
+char *am_str_join(apr_pool_t *pool, char **strings, const char *separator)
+{
+    char *str = NULL;
+    char *dst = NULL;
+    char **p;
+    apr_size_t i;
+    apr_size_t len;
+    apr_size_t alloc_len;
+    apr_size_t n_strings = 0;
+    apr_size_t total_string_len = 0;
+    apr_size_t separator_len = strlen(separator);
+
+    /*
+     * Iterate over string array, compute total number of characters
+     * and a count of how many strings
+     */
+    for (total_string_len = 0, n_strings = 0, p = strings; *p; p++) {
```

Comment:
I think this loop would be much clearer if the initialization of `total_string_len` and `n_strings` happened outside of the for-loop. Including them in the for-loop like this seems to imply that it is something it iterates over, rather than what it actually counts.